profiler/options.go

```diff
@@ -100,6 +101,44 @@ type config struct {
 	blockRate         int
 	outputDir         string
 	deltaProfiles     bool
+	logStartup        bool
+}
+
+// logStartup records the configuration to the configured logger in JSON format
+func logStartup(c *config) {
+	info := struct {
```

Comment:
NIT: The tracer includes some additional fields that could be useful, e.g. go version, os version, os name, etc., see:

https://github.com/DataDog/dd-trace-go/blob/b6dbd9f0897d8152d5c802acb77ecabb738222ed/ddtrace/tracer/log.go#L84-L89

Maybe we should log those as well?

Additionally the following fields from the config struct seem missing:

- hostname
- deltaProfiles
- maxGoroutinesWait

Reply:
Thanks for the review. I added the missing config fields and checked for the startup log in a test.

RE: the other fields, would it make sense to pull the osName & osVersion functions from the tracer package into their own internal package so the implementations can be shared?

Reply:
OK, I pulled out the OS info to its own package and added the other fields for the profiler log. PTAL